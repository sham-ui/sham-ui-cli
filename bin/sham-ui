#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .parse( process.argv );