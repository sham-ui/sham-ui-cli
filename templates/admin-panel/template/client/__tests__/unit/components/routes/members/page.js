import { DI } from 'sham-ui';
import * as directives from 'sham-ui-directives';
import RoutesMembersPage  from '../../../../../src/components/routes/members/page.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    DI.resolve( 'session:storage' ).sessionValidated = true;

    const getMock = jest.fn();
    DI.bind( 'store', {
        api: {
            request: getMock.mockReturnValueOnce(
                Promise.resolve( { } )
            )
        }
    } );

    const meta = renderer( RoutesMembersPage, {
        directives
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'display errors', async() => {
    DI.resolve( 'session:storage' ).sessionValidated = true;
    DI.bind( 'store', {
        api: {
            request: jest.fn().mockReturnValueOnce(
                Promise.reject( {} )
            )
        }
    } );
    const meta = renderer( RoutesMembersPage, {
        directives
    } );
    await new Promise( resolve => setImmediate( resolve ) );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
