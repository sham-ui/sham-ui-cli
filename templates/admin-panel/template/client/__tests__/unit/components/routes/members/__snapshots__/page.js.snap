// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display errors 1`] = `
Object {
  "Constructor": "RoutesMembersPage",
  "Options": Object {
    "ID": "component",
    "dataLoaded": true,
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [
      "Fail load members list",
    ],
    "limit": 50,
    "members": Array [],
    "offset": 0,
    "total": 0,
  },
  "html": "
<h4 class=\\"page-header\\"> Members
  <!--header-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"member-list\\">
    <ul class=\\"errors-panel\\">
      <li>Fail load members list</li>
    </ul>
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"panel\\"></div>
  </div>
  <!--body-->
</div>
<!--if-->
<!--Page-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Constructor": "RoutesMembersPage",
  "Options": Object {
    "ID": "component",
    "dataLoaded": false,
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [],
    "limit": 50,
    "members": Array [],
    "offset": 0,
    "total": 0,
  },
  "html": "
<h4 class=\\"page-header\\"> Members
  <!--header-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"member-list\\">
    <!--if-->
    <!--ErrorsPanel-->
    <div class=\\"panel\\"></div>
  </div>
  <!--body-->
</div>
<!--if-->
<!--Page-->
",
}
`;
